sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParserTest.java

```diff
@@ -129,16 +130,56 @@ public void branchCoverage() throws Exception {
         Assertions.entry(18, 1),
         Assertions.entry(25, 1));
 
-    assertThat(coverage.getBranchCoverage(filePath).size()).isEqualTo(5);
+    assertThat(coverage.getBranchCoverage(filePath))
+      .hasSize(5)
+      .contains(
+        new BranchCoverage(9, 4 , 2),
+        new BranchCoverage(12, 2 , 1),
+        new BranchCoverage(13, 2, 1),
+        new BranchCoverage(14, 4, 2),
+        new BranchCoverage(18, 6, 3));
+  }
+
+  @Test
+  public void branchCoverage_codeFile_analyzedByMultipleProjects() throws Exception {
+    Coverage coverage = new Coverage();
+    String filePath = new File("BranchCoverage3296\\Code\\ValueProvider.cs").getCanonicalPath();
+
+    new OpenCoverReportParser(alwaysTrue).accept(new File("src/test/resources/opencover/code_tested_by_multiple_projects.xml"), coverage);
+    assertThat(coverage.files()).containsOnly(filePath);
+
+    assertThat(coverage.getBranchCoverage(filePath)).containsOnly(new BranchCoverage(5, 2, 2));
+  }
+
+  @Test
+  public void branchCoverage_codeFile_unsupportedFile() throws Exception {
+    Coverage coverage = new Coverage();
+    String filePath = new File("BranchCoverage3296\\Code\\ValueProvider.cs").getCanonicalPath();
+
+    new OpenCoverReportParser(alwaysFalse).accept(new File("src/test/resources/opencover/code_tested_by_multiple_projects.xml"), coverage);
```

Comment:
I would add a comment above like `// Notice we pass "alwaysFalse" as a predicate`

Reply:
this is basically the only difference between this and the above test and it's easy to miss